app.py

```
@@ -0,0 +1,81 @@
+
+    if request.method == "POST":
+        data = request.json
+        lat, lon = data.get("lat"), data.get("lon")
+    else:
+        city = request.args.get("city")
+        lat, lon = request.args.get("lat"), request.args.get("lon")
+
+    if city and not (lat and lon):
+        geo_url = f"http://api.openweathermap.org/geo/1.0/direct?q={city}&limit=1&appid={API_KEY}"
+        try:
+            geo_res = requests.get(geo_url).json()
```

Ruff (0.15.2)

[error] Probable use of `requests` call without timeout

(S113)